drivers/xen/gnttab.c

```diff
@@ -306,8 +311,18 @@ static int gnttab_init(const struct device *d)
 		put_free_entry(gref);
 	}
 
-	gnttab.table = (grant_entry_v1_t *)
-			DT_REG_ADDR_BY_IDX(DT_INST(0, xen_xen), 0);
+	/*
+	 * User to internal flag conversions:
+	 * K_MEM_CACHE_WB  --> MT_NORMAL
+	 * K_MEM_PERM_RW   --> MT_RW
+	 * K_MEM_PERM_EXEC --> MT_P_EXECUTE_NEVER
```

Comment:
Wow! Are you sure that `K_MEM_PERM_EXEC` maps to `MT_P_EXECUTE_NEVER`? This is very strange. Those are two opposite values.

Actually, `__arch_mem_map()` confirms me:

```
	if ((flags & K_MEM_PERM_EXEC) == 0U) {
		entry_flags |= MT_P_EXECUTE_NEVER;
	}
```

Reply:
shame on me